Enhancement: Display inactive-account indicator next to locked accounts at various locations

systemofapwne opened the pull request:
<!--

Thanks for contributing to the Urlaubsverwaltung.
Please review the following notes before submitting you pull request.

Please look for other issues or pull requests which already work on this topic. Is somebody already on it? Do you need to synchronize?

# Security Vulnerabilities

🛑 STOP! 🛑 If your contribution fixes a security vulnerability, please do not submit it.
Instead, please write an E-Mail to info@urlaubsverwaltung.cloud with all the information
to recreate the security vulnerability.

# Describing Your Changes

If, having reviewed the notes above, you're ready to submit your pull request, please
provide a brief description of the proposed changes.

If they:
 🐞 fix a bug, please describe the broken behaviour and how the changes fix it.
    Please label with 'type: bug' and 'status: new'
    
 🎁 make an enhancement, please describe the new functionality and why you believe it's useful.
    Please label with 'type: enhancement' and 'status: new'
 
If your pull request relates to any existing issues,
please reference them by using the issue number prefixed with #.

-->

This PR adds a lock-symbol next to accounts, that have been set to "INACTIVE".
The goal is to help making sense about (deprecated) applications & leaves of users, that have left the company or are deactivated for other reasons,

The symbol will be visible at the following locations:
- Person Overview
- Person Detail View
  - Person Base Data Editor
  - Person Permission Editor
- Application Overview

derTobsch commented:
Hey @systemofapwne,

thanks for your PR! I like the idea to visualize that a person is in active. Some thoughts of mine.

* The icon should not be in a link
* We need to unify the "inactive" icons
* A11y should be added and maybe a title or some information what the lock stands for
* Lock should be from lucide (and it is or? https://lucide.dev/icon/lock?search=lock)
* Dark mode (seems ok)

Unification:
Persons and other places? the eye-off icon is only used here
![grafik](https://user-images.githubusercontent.com/5683677/214572939-b14e607a-1106-49cb-8840-8a977cccfc12.png)

systemofapwne commented:
> Hey @systemofapwne,
> 
> thanks for your PR! I like the idea to visualize that a person is in active. Some thoughts of mine.
> 
> * The icon should not be in a link
> * We need to unify the "inactive" icons
> * A11y should be added and maybe a title or some information what the lock stands for
> * Lock should be from lucide (and it is or? https://lucide.dev/icon/lock?search=lock)
> * Dark mode (seems ok)
> 
> Unification: Persons and other places? the eye-off icon is only used here ![grafik](https://user-images.githubusercontent.com/5683677/214572939-b14e607a-1106-49cb-8840-8a977cccfc12.png)
> 
> @bseber what do you think?

Thank you for your reply.

Alright, I will move the icon out of the link. That is a no-brainer.
I will also change the icon from the lock-type to the eye-type to unify the icon-style and add a hover-message to it.
The lock-icon was from lucide.

I did not add the indicator to other locations yet but I agree, this makes sense.

Btw: Right now, I am struggling with the automated tests, that seem to fail on github and also on my pc on a completely unrelated location to my changes: 
``
ERROR] Errors: 
[ERROR]   OvertimeCreateIT.ensureOvertimeCreation:138 » Timeout Expected condition faile...
``
which can be traced back up to here: https://github.com/urlaubsverwaltung/urlaubsverwaltung/blob/886a78ecbd5d299b4dd4a4af107b86def8b70bd8/src/test/java/org/synyx/urlaubsverwaltung/ui/SickNoteIT.java#L144

If you could direct me towards the error, this would be great. 
Thanks in advance.

systemofapwne commented:
I added the requested changes and fixed the "test" run, that initially failed.

A11y: Can you please elaborate, if the added "title" tag in the span around the SVGs is sufficient?

bseber commented:
nice one @systemofapwne 🤩

a11y: I would just add plain text which could be (maybe should be in our case) hidden visually. 

```html
Jane Doe <small><svg ... /><span class="tw-sr-only">(inactive user)</span></small>
```

Regarding the icon: Introducing this element in the whole application we should think about whether the _crossed eye_ is ok or not. I think the _eye_ could be useful for other use cases whether something should be visible or hidden. The user state "disabled" / "inactive" could be communicated with:

eye:

![inactive_eye](https://user-images.githubusercontent.com/1732200/215038979-86e7b792-1f22-4c85-8cd2-905b6b045b14.png)

ghost:

![incative_ghost](https://user-images.githubusercontent.com/1732200/215038973-1b242436-3d1b-475c-ae12-098fb54e506b.png)

user-x:

![inactive_person-x](https://user-images.githubusercontent.com/1732200/215038977-a173bbd8-f132-4d9b-ab21-d100684d6373.png)

octagon:

![inactive_octagon](https://user-images.githubusercontent.com/1732200/215038968-5a2b09f7-c48a-43ce-8eac-d8dfb447a8ec.png)



I quite like the _ghost_ icon as it is kind of funny and arouses emotions. Second vote goes to _user-x_.

derTobsch commented:
> nice one @systemofapwne star_struck
> 
> a11y: I would just add plain text which could be (maybe should be in our case) hidden visually.
> 
> ```
> Jane Doe <small><svg ... /><span class="tw-sr-only">(inactive user)</span></small>
> ```
> 
> Regarding the icon: Introducing this element in the whole application we should think about whether the _crossed eye_ is ok or not. I think the _eye_ could be useful for other use cases whether something should be visible or hidden. The user state "disabled" / "inactive" could be communicated with:
> 
> eye:
> 
> ![inactive_eye](https://user-images.githubusercontent.com/1732200/215038979-86e7b792-1f22-4c85-8cd2-905b6b045b14.png)
> 
> ghost:
> 
> ![incative_ghost](https://user-images.githubusercontent.com/1732200/215038973-1b242436-3d1b-475c-ae12-098fb54e506b.png)
> 
> user-x:
> 
> ![inactive_person-x](https://user-images.githubusercontent.com/1732200/215038977-a173bbd8-f132-4d9b-ab21-d100684d6373.png)
> 
> octagon:
> 
> ![inactive_octagon](https://user-images.githubusercontent.com/1732200/215038968-5a2b09f7-c48a-43ce-8eac-d8dfb447a8ec.png)
> 
> I quite like the _ghost_ icon as it is kind of funny and arouses emotions. Second vote goes to _user-x_.

but if we use a new icon, than we should use it everywhere in the application. :)

systemofapwne commented:
> nice one @systemofapwne star_struck
> 
> a11y: I would just add plain text which could be (maybe should be in our case) hidden visually.
> 
> ```
> Jane Doe <small><svg ... /><span class="tw-sr-only">(inactive user)</span></small>
> ```
> 
> Regarding the icon: Introducing this element in the whole application we should think about whether the _crossed eye_ is ok or not. I think the _eye_ could be useful for other use cases whether something should be visible or hidden. The user state "disabled" / "inactive" could be communicated with:
> 
> eye:
> 
> https://user-images.githubusercontent.com/1732200/215038979-86e7b792-1f22-4c85-8cd2-905b6b045b14.png
> 
> ghost:
> 
> https://user-images.githubusercontent.com/1732200/215038973-1b242436-3d1b-475c-ae12-098fb54e506b.png
> 
> user-x:
> 
>https://user-images.githubusercontent.com/1732200/215038977-a173bbd8-f132-4d9b-ab21-d100684d6373.png
> 
> octagon:
> 
> https://user-images.githubusercontent.com/1732200/215038968-5a2b09f7-c48a-43ce-8eac-d8dfb447a8ec.png
> 
> I quite like the _ghost_ icon as it is kind of funny and arouses emotions. Second vote goes to _user-x_.


Thank you. I added your suggestion regarding A11y.

Regarding the icon decision, I am open for everything. Right now, the eye-off next to the name looks a bit out of place. 
This is not a big deal for the users list, since the text "active/inactive" is written right next to it.
So a decision has to be made: 
- We stick to the current symbols and accept, that eye-off standing alone is not so conclusive.
- Or we introduce a new icon set of active & inactive, that are conclusive even when standing alone.

derTobsch commented:
> > nice one @systemofapwne star_struck
> > a11y: I would just add plain text which could be (maybe should be in our case) hidden visually.
> > ```
> > Jane Doe <small><svg ... /><span class="tw-sr-only">(inactive user)</span></small>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Regarding the icon: Introducing this element in the whole application we should think about whether the _crossed eye_ is ok or not. I think the _eye_ could be useful for other use cases whether something should be visible or hidden. The user state "disabled" / "inactive" could be communicated with:
> > eye:
> > https://user-images.githubusercontent.com/1732200/215038979-86e7b792-1f22-4c85-8cd2-905b6b045b14.png
> > ghost:
> > https://user-images.githubusercontent.com/1732200/215038973-1b242436-3d1b-475c-ae12-098fb54e506b.png
> > user-x:
> > https://user-images.githubusercontent.com/1732200/215038977-a173bbd8-f132-4d9b-ab21-d100684d6373.png
> > octagon:
> > https://user-images.githubusercontent.com/1732200/215038968-5a2b09f7-c48a-43ce-8eac-d8dfb447a8ec.png
> > I quite like the _ghost_ icon as it is kind of funny and arouses emotions. Second vote goes to _user-x_.
> 
> Thank you. I added your suggestion regarding A11y.
> 
> Regarding the icon decision, I am open for everything. Right now, the eye-off next to the name looks a bit out of place. This is not a big deal for the users list, since the text "active/inactive" is written right next to it. So a decision has to be made:
> 
>     * Either we stick to the current symbols and accept, that eye-off standing alone is not so conclusive.
> 
>     * Or we introduce a new icon set of active & inactive, that are conclusive even when standing alone.
> 
> 
> If we choose a new icon-set, user-x and a complementary icon would be my favorites too.

My first thoughts are.

* Replace eye with person
* Replace eye-off with person-x

and we should change in all places. What do you mean @bseber ?

systemofapwne commented:
I arbitrarily decided on changing the indicator to user and user-x everywhere. Feel free to comment on this decision. I am open for everything.

derTobsch commented:
Thanks @systemofapwne !
